### PR TITLE
Migrate to using mucs for jvb

### PR DESCRIPTION
**Pull request by damencho**
Not sure which one to merge first, I think tests that test the debian package deployment will fail till we have all 3 components jitsi-meet, jicofo and jvb changes merged.
Once this is merged I will cherry-pick it for jvb2.

https://github.com/jitsi/jitsi-meet/pull/4956
https://github.com/jitsi/jicofo/pull/413

### CODE REVIEW

File: lib/logging.properties [hunk 1]
```diff
@@ -23,3 +23,5 @@ com.agafua.syslog.SyslogHandler.escapeNewlines = false
 
 # to disable double timestamps in syslog uncomment next line
 #net.java.sip.communicator.util.ScLogFormatter.disableTimestamp=true
+
+org.jitsi.xmpp.mucclient.MucClientManager.level=SEVERE
```
Comment (bgrozev) [inline lib/logging.properties]: Is this to silence "Setting a presence extension" logs?

File: resources/install/debian/postinst [hunk 2]
```diff
@@ -96,8 +96,23 @@ case "$1" in
         # and this breaks startup script
         sed -i 's/\$JVB_EXTRA_JVM_PARAMS//g' $CONFIG
 
-        if ! grep -q "org.jitsi.videobridge.AUTHORIZED_SOURCE_REGEXP" "$NEW_JITSI_CONFIG" ;then
-            echo "org.jitsi.videobridge.AUTHORIZED_SOURCE_REGEXP=focus@auth.${JVB_HOSTNAME}/.*" >> $NEW_JITSI_CONFIG
+        if ! grep -q "#org.jitsi.videobridge.AUTHORIZED_SOURCE_REGEXP" "$NEW_JITSI_CONFIG" ;then
```
Comment (bgrozev) [inline resources/install/debian/postinst]: Is the `if` necessary? The only case where it makes a difference is when the file contains both the commented-out and non-commented-out version.

Does this script execute when the debian package is updated?

File: resources/install/debian/postinst [hunk 3]
```diff
@@ -96,8 +96,23 @@ case "$1" in
...
+
+        if ! grep -q "org.jitsi.videobridge.xmpp.user.shard.MUC_JIDS" "$NEW_JITSI_CONFIG" ;then
+            echo "org.jitsi.videobridge.ENABLE_STATISTICS=true" >> $NEW_JITSI_CONFIG
+            echo "org.jitsi.videobridge.STATISTICS_TRANSPORT=muc" >> $NEW_JITSI_CONFIG
+            echo "org.jitsi.videobridge.xmpp.user.shard.HOSTNAME=localhost" >> $NEW_JITSI_CONFIG
+            echo "org.jitsi.videobridge.xmpp.user.shard.DOMAIN=auth.$JVB_HOSTNAME" >> $NEW_JITSI_CONFIG
+            echo "org.jitsi.videobridge.xmpp.user.shard.USERNAME=jvb" >> $NEW_JITSI_CONFIG
+            echo "org.jitsi.videobridge.xmpp.user.shard.PASSWORD=$JVB_SECRET" >> $NEW_JITSI_CONFIG
+            echo "org.jitsi.videobridge.xmpp.user.shard.MUC_JIDS=JvbBrewery@internal.auth.$JVB_HOSTNAME" >> $NEW_JITSI_CONFIG
+            echo "org.jitsi.videobridge.xmpp.user.shard.MUC_NICKNAME=$(uuidgen)" >> $NEW_JITSI_CONFIG
```
Comment (bgrozev) [inline resources/install/debian/postinst]: I don't think we need anything unique here, just "jvb" will do.

File: resources/install/debian/postinst [hunk 4]
```diff
@@ -56,7 +56,7 @@ case "$1" in
             echo "JVB_SECRET=$JVB_SECRET" >> $CONFIG
             echo >> $CONFIG
             echo '# extra options to pass to the JVB daemon' >> $CONFIG
-            echo "JVB_OPTS=\"\"" >> $CONFIG
+            echo "JVB_OPTS=\"--apis=rest\"" >> $CONFIG
```
Comment (bgrozev) [inline resources/install/debian/postinst]: Does this change the default value of the rest interface?

File: lib/logging.properties (hunk 1 above)
Comment (damencho) [reply lib/logging.properties]: Exactly

File: resources/install/debian/postinst (hunk 2 above)
Comment (damencho) [reply resources/install/debian/postinst]: Yes, this is executed on every update. The goal of this is to make current configs work(migrate them) and give people an option to anyway have something in there and we do not override it on update(having commented and not commented line). It is not documented but it is an option if someone complains about something like that.

File: resources/install/debian/postinst (hunk 3 above)
Comment (damencho) [reply resources/install/debian/postinst]: What about adding a second jvb. You can do the standard installation on one machine(jitsi-meet) and on a second one just install jvb ...

File: resources/install/debian/postinst (hunk 4 above)
Comment (damencho) [reply resources/install/debian/postinst]: Well, the intention is to turn off the component, as if we leave it, it fails health check. 
When I was initially adding the mucs support I added --no-component param for jigasi and jvb, but I think we merged only the jigasi part.

File: resources/install/debian/postinst (hunk 2 above)
Comment (bgrozev) [reply resources/install/debian/postinst]: Okay..

File: resources/install/debian/postinst (hunk 3 above)
Comment (bgrozev) [reply resources/install/debian/postinst]: I thought this required manual configuration anyway. I don't mind using a uuid

File: resources/install/debian/postinst (hunk 4 above)
Comment (bgrozev) [reply resources/install/debian/postinst]: My question is about the rest interface, not the component. Does a default debian installation come with rest enabled or disabled? It looks like it used to be disabled, and now it's enabled.

Comment (damencho) [reply resources/install/debian/postinst]: Yeah, that's true ... I'm not sure how to disable the component another way. I will take a look.

Comment (bgrozev) [reply resources/install/debian/postinst]: I think just passing `--apis=` should do it

File: lib/logging.properties (hunk 1 above)
Comment (bbaldino) [reply lib/logging.properties]: Should we move that log to a lower level instead?

Comment (bgrozev) [reply lib/logging.properties]: We could do it in addition. This PR targets `jvb_1.0` and can't use an updated jicofo (unless we maintain a branch, but that would be a lot of hassle to silence some logs)

File: resources/install/debian/postinst (hunk 4 above)
Comment (damencho) [reply resources/install/debian/postinst]: > I think just passing `--apis=` should do it

@bgrozev that doesn't work

```
JVB 2020-01-07 17:46:15.489 SEVERE: [33] org.jitsi.meet.ComponentMain.log() host-unknown, host:localhost, port:5347
org.xmpp.component.ComponentException: host-unknown
	at org.jivesoftware.whack.ExternalComponent.connect(ExternalComponent.java:243)
```

Comment (damencho) [reply resources/install/debian/postinst]: Hum, but `--apis=,` works :) I will change it to that

Comment (bgrozev) [reply resources/install/debian/postinst]: Nice find :)

**Pull request merged**